uefi/src/data_types/strs.rs

```diff
@@ -309,26 +309,47 @@ impl CStr16 {
         })
     }
 
-    /// Returns the inner pointer to this C string
+    /// Returns the inner pointer to this C16 string.
     #[must_use]
     pub const fn as_ptr(&self) -> *const Char16 {
         self.0.as_ptr()
     }
 
-    /// Get the underlying [`Char16`] slice, including the trailing null.
+    /// Get the underlying [`Char16`]s as slice without the trailing null.
+    #[must_use]
+    pub fn as_slice(&self) -> &[Char16] {
+        &self.0[..self.len()]
+    }
+
+    /// Get the underlying [`Char16`]s as mut slice without the trailing null.
+    #[must_use]
+    fn as_slice_mut(&mut self) -> &mut [Char16] {
+        let len = self.len();
+        &mut self.0[..len]
+    }
+
+    /// Replaces all chars in the string with the replace value.
+    pub fn char_replace_all_in_place(&mut self, search: Char16, replace: Char16) {
```

Comment:
Having a method of `CStr16` take `&mut self` feels a little surprising since we pretty much always have a `&CStr16`, not `&mut CStr16`. For comparison the `str` replace method (https://doc.rust-lang.org/std/primitive.str.html#method.replace) doesn't modify in-place, but returns a new `String`. We could just leave this method on `CString16` and drop it from `CStr16`?